RefrigeratorSaraFarkash/RefrigeratorSaraFarkash/CRefrigerator.cs

```diff
@@ -0,0 +1,334 @@
+        {
+            List<CItem> wanttoeat = new List<CItem>();
+
+            for (int i = 0; i < shelfs.Count; i++)
+            {
+                foreach (CItem item in shelfs[i].Items)
+                {
+                    if (item.Kashroot == kashroot && item.TypeI == typeeat && item.ExpiryDate > DateTime.Today)
+                        wanttoeat.Add(item);
+                }
+            }
+            return wanttoeat;
+        }
+
+        // sort
+
+        public int CompareTo(CRefrigerator otherRefrigerator)
+        {
+            return ((this.spaceLeftInRefrigerator() > otherRefrigerator.spaceLeftInRefrigerator()) ? (-1) : (this.spaceLeftInRefrigerator() == otherRefrigerator.spaceLeftInRefrigerator()) ? 0 : 1);
+        }
+
+        public List<CItem> SortItems()
+        {
+            List<CItem> items = new List<CItem>();
+            foreach (CShelf shelf in this.shelfs)
+            {
+                items.AddRange(shelf.Items);
+            }
+            items.Sort();
+            return items;
+        }
+        public List<CShelf> SortShelf()
+        {
+            List<CShelf> shelfs = new List<CShelf>();
+            foreach (CShelf shelf in this.shelfs)
+            {
+                shelfs.Add(shelf);
+            }
+            shelfs.Sort();
+
+            return shelfs;
+        }
+        //1
+        //
+        //A function that receives a date and returns an list with all the food deleted up to the date
+        public List<CItem> wantToREMOVE(int untilday, string KashrootItem)
+        {
+            List<CItem> RemoveItem = new List<CItem>();
+            DateTime lastDate = DateTime.Today;
+            lastDate = lastDate.AddDays(untilday);
+            for (int i = 0; i < this.numberOfShelves; i++)
+            {
+                foreach (CItem item in this.Shelfs[i].Items)
+                {
+                    if (item.ExpiryDate <= lastDate && KashrootItem == item.Kashroot)
+                    {
+                        RemoveItem.Add(item);
+                    }
+                }
+                foreach (CItem itemToRemove in RemoveItem)
+                {
+                    CItem itemremving = removeItemFromRef(itemToRemove.Id);
+                }
+            }
+            return RemoveItem;
+        }
+        public Boolean ThereIsPlacefor20()
+        {
+            if (this.spaceLeftInRefrigerator() >= 20)
+                return true;
+            return false;
+        }
+
+        public void addListItems(List<CItem> itemsAdd)
+        {
+            foreach (CItem item in itemsAdd)
+            {
+                this.pushToRefItem(item);
+            }
+        }
+
+        public void beforeShopping()
+        {
+
+            List<CItem> dairyRemove = new List<CItem>();
+            List<CItem> meatyRemove = new List<CItem>();
+            List<CItem> parveRemove = new List<CItem>();
+            int flag = 0;
+            if (this.spaceLeftInRefrigerator() < 29)
```

Comment:
The 29 in the document was probably a type.
It is OK to ask questions when the document contradicts itself.
Don't worry this will not count as a mistake